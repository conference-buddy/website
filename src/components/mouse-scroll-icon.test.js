import React from 'react'
import { shallow } from 'enzyme'
import MouseScrollIcon from './mouse-scroll-icon'

const testOnPageLinkId = 'anchor-target'

describe('<MouseScrollIcon />', () => {
  it('renders without crashing', () => {
    shallow(<MouseScrollIcon onPageLinkId={testOnPageLinkId} />)
  })

  describe('renders a link to navigate onPage', () => {
    const wrapper = shallow(<MouseScrollIcon onPageLinkId={testOnPageLinkId} />)

    it('rendes a link an css icon as content', () => {
      const icon = wrapper.find('.MouseScrollIcon-mouse')
      expect(icon).toHaveLength(1)
      expect(icon).toMatchSnapshot()
    })
    it('renders a link with an onClick function', () => {
      expect(wrapper.find('a').prop('href')).toEqual(testOnPageLinkId)
      expect(wrapper.find('a').prop('onClick')).toHaveLength(1)
    })
    it('renders a link with an onClick function', () => {
      expect(wrapper.find('a').prop('href')).toEqual(testOnPageLinkId)
      expect(wrapper.find('a').prop('onClick')).toHaveLength(1)
    })
  })
})
