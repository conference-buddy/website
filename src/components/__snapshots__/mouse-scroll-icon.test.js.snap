// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MouseScrollIcon /> renders a link to navigate onPage rendes a link an css icon as content 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <MouseScrollIcon
      onPageLinkId="anchor-target"
    />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          href="anchor-target"
          onClick={[Function]}
        >
          <span
            className="MouseScrollIcon-mouse"
          >
            <span
              className="MouseScrollIcon-wheel"
            />
          </span>
        </a>,
        "className": "MouseScrollIcon d-none d-sm-block",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="MouseScrollIcon-mouse"
          >
            <span
              className="MouseScrollIcon-wheel"
            />
          </span>,
          "href": "anchor-target",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="MouseScrollIcon-wheel"
            />,
            "className": "MouseScrollIcon-mouse",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "MouseScrollIcon-wheel",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "span",
        },
        "type": "a",
      },
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="anchor-target"
            onClick={[Function]}
          >
            <span
              className="MouseScrollIcon-mouse"
            >
              <span
                className="MouseScrollIcon-wheel"
              />
            </span>
          </a>,
          "className": "MouseScrollIcon d-none d-sm-block",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="MouseScrollIcon-mouse"
            >
              <span
                className="MouseScrollIcon-wheel"
              />
            </span>,
            "href": "anchor-target",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                className="MouseScrollIcon-wheel"
              />,
              "className": "MouseScrollIcon-mouse",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "MouseScrollIcon-wheel",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "span",
          },
          "type": "a",
        },
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
      "attachTo": undefined,
      "hydrateIn": undefined,
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="MouseScrollIcon-wheel"
      />,
      "className": "MouseScrollIcon-mouse",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "MouseScrollIcon-wheel",
      },
      "ref": null,
      "rendered": null,
      "type": "span",
    },
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="MouseScrollIcon-wheel"
        />,
        "className": "MouseScrollIcon-mouse",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "MouseScrollIcon-wheel",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
